src/ParallelAlgorithms/Interpolation/Callbacks/ObserveTimeSeriesOnSurface.hpp

```diff
@@ -40,9 +40,34 @@ namespace callbacks {
 
 namespace detail {
 
+template<typename T>
+struct is_array_of_double : std::false_type {};
+
+template<std::size_t N>
+struct is_array_of_double<std::array<double, N>> : std::true_type {};
+
 template <typename... Ts>
 auto make_legend(tmpl::list<Ts...> /* meta */) {
-  return std::vector<std::string>{"Time", db::tag_name<Ts>()...};
+    std::vector<std::string> legend = {"Time"};
+    
+    auto append_tags = [&legend](auto tag) {
+        using TagType = decltype(tag);
+        using ReturnType = typename TagType::return_type;
+
+        if constexpr (is_array_of_double<ReturnType>::value) {
+            for (size_t i = 0; i < std::tuple_size<ReturnType>::value; ++i) {
+                legend.push_back(db::tag_name<TagType>() + std::to_string(i));
```

Comment:
Please output these as `TagName_x`, `TagName_y`, `TagName_z`. I understand that the code is generic for any number of array elements and they don't necessarily mean dimensions, but for current use cases they do and it's important to have the "xyz" suffix in the output I think for usability of the output data.

(In the future I think we should change the type of the `DimensionfulSpinVector` tag to `tnsr::I<double, 3, Frame::Inertial>` so it's unambiguous what the numbers mean and in which coordinate frame they are, but that's out of scope for this PR)